packages/pr-antora-content-guidelines-checker/src/main.ts

```diff
@@ -0,0 +1,99 @@
+import * as core from "@actions/core";
+import * as github from "@actions/github";
+import { ActionResult, Status, ValidationStep } from "./validation";
+import {
+  getModifiedFiles,
+  publishComment,
+  isCommentExist,
+  deleteComment,
+} from "./github-utils";
+import { AttributesCheckingStep } from "./steps/AttributesCheckingStep";
+import { ForbiddenPatternStep } from "./steps/ForbiddenPatternStep";
+import { GitHub } from "@actions/github/lib/utils";
+
+const template = "<!-- previewCommentContributionChecker -->\n";
+
+async function run(): Promise<void> {
+  try {
+    const token = core.getInput("github-token");
+    const octokit: InstanceType<typeof GitHub> = github.getOctokit(token);
+    let actionResult: ActionResult[] = [];
+    const modifiedFiles: string[] = await getModifiedFiles(octokit);
+
+    const filesToCheckInput = core.getInput("files-to-check").split(",");
+    const attributesToCheckInput = core
+      .getInput("attributes-to-check")
+      .split(",");
+    const forbiddenPatternToCheckInput = core
+      .getInput("forbidden-pattern-to-check")
+      .split(",");
+    let steps: Array<ValidationStep> = [];
+
+    if (core.getInput("attributes-to-check") !== "") {
+      steps.push(
+        new AttributesCheckingStep(
+          modifiedFiles,
+          filesToCheckInput,
+          attributesToCheckInput
+        )
+      );
+    }
+    if (core.getInput("forbidden-pattern-to-check") !== "") {
+      steps.push(
+        new ForbiddenPatternStep(
+          modifiedFiles,
+          filesToCheckInput,
+          forbiddenPatternToCheckInput
+        )
+      );
+    }
+
+    core.info("Input parameters:");
+    core.info(`* files-to-check: ${filesToCheckInput.join(", ")}`);
+    core.info(`* attributes-to-check: ${attributesToCheckInput}`);
+    core.info(`* forbidden-pattern-to-check: ${forbiddenPatternToCheckInput}`);
+
+    for (const step of steps) {
+      core.debug(`------- ${step.name} -------`);
+      let stepResult = await step.validate(octokit, modifiedFiles);
+      actionResult.push(stepResult);
+    }
+
+    core.setOutput("checker-result", actionResult);
+
+    const filterResultOnError = actionResult.filter(
+      (result) => result.status === Status.ERROR
+    );
+
+    const prNumber = github?.context?.payload?.pull_request?.number;
+    if (filterResultOnError.length >= 1) {
+      core.setFailed(
+        `This PR didn't following all guideline, check the comments to see more details`
+      );
+      if (prNumber) {
+        let commentBody: string = template + "# PR Guideline checker\n";
```

Comment:
**suggestion:** what do you think of these alternatives for the title of the comment?
- Contribution Guidelines checks
- verification of contribution guidelines
- Contribution guidelines checker

We could also add a short sentence right after the title to explain what is described in the comment. For example: 
> The content of the files modified by this Pull Request doesn't match the _Contribution Guidelines_.
Please update the following files.

Reply:
done !